diff --git a/compilationOptions.original b/cmake/CompilationOptions.cmake
index bb60c31..d8a2df5 100644
--- a/compilationOptions.original
+++ b/cmake/CompilationOptions.cmake
@@ -47,7 +47,7 @@ option(DOUBLE_PRECISION "Build Amber's Fortran programs with double precision ma
 
 #let's try to enforce a reasonable standard here
 set(CMAKE_C_STANDARD 99)
-set(CMAKE_CXX_STANDARD 11)
+set(CMAKE_CXX_STANDARD 14)
 
 # I can't think of any better place to put this...
 
@@ -133,4 +133,4 @@ if("${CMAKE_C_COMPILER_ID}" STREQUAL "PGI" AND HOST_LINUX AND "${TARGET_ARCH}" S
 	list(APPEND CMAKE_INCLUDE_PATH /usr/include/x86_64-linux-gnu)
 	list(APPEND CMAKE_LIBRARY_PATH /usr/lib/x86_64-linux-gnu)
 
-endif()
\ No newline at end of file
+endif()
