diff -Nau wmake/rules/linux64Cray/c wmake/rules/linux64CrayToBeAdded/c
--- wmake/rules/linux64Cray/c	1970-01-01 02:00:00.000000000 +0200
+++ wmake/rules/linux64CrayToBeAdded/c	2025-02-25 13:28:52.000000000 +0200
@@ -0,0 +1,18 @@
+SUFFIXES += .c
+
+cWARN        = -Wall
+
+cc          = cc -m64
+
+include $(DEFAULT_RULES)/c$(WM_COMPILE_OPTION)
+
+cFLAGS      = $(GFLAGS) $(cWARN) $(cOPT) $(cDBUG) $(LIB_HEADER_DIRS) -fPIC
+
+ctoo        = $(WM_SCHEDULER) $(cc) $(cFLAGS) -c $< -o $@
+
+LINK_LIBS   = $(cDBUG)
+
+LINKLIBSO   = $(cc) -fuse-ld=bfd -shared
+
+LINKEXE     = $(cc) -fuse-ld=bfd \
+              -Xlinker --add-needed -Xlinker -z -Xlinker nodefs
diff -Nau wmake/rules/linux64Cray/c++ wmake/rules/linux64CrayToBeAdded/c++
--- wmake/rules/linux64Cray/c++	1970-01-01 02:00:00.000000000 +0200
+++ wmake/rules/linux64CrayToBeAdded/c++	2025-02-25 13:28:52.000000000 +0200
@@ -0,0 +1,29 @@
+SUFFIXES += .C
+
+c++WARN     = -Wall -Wextra -Wold-style-cast -Wnon-virtual-dtor \
+              -Wno-unused-parameter -Wno-invalid-offsetof -Wno-attributes
+
+# Suppress some warnings for flex++
+c++LESSWARN = -Wno-old-style-cast -Wno-unused-local-typedefs -Wno-array-bounds
+
+CC          = CC -std=c++14 -m64
+
+include $(DEFAULT_RULES)/c++$(WM_COMPILE_OPTION)
+
+ptFLAGS     = -DNoRepository -ftemplate-depth-100
+
+c++FLAGS    = $(GFLAGS) $(c++WARN) $(c++OPT) $(c++DBUG) $(ptFLAGS) \
+              $(LIB_HEADER_DIRS) -fPIC
+
+Ctoo        = $(WM_SCHEDULER) $(CC) $(c++FLAGS) -c $< -o $@
+cxxtoo      = $(Ctoo)
+cctoo       = $(Ctoo)
+cpptoo      = $(Ctoo)
+
+LINK_LIBS   = $(c++DBUG)
+
+LINKLIBSO   = $(CC) $(c++FLAGS) -fuse-ld=bfd -shared \
+              -Xlinker --add-needed -Xlinker --no-as-needed
+
+LINKEXE     = $(CC) $(c++FLAGS) -fuse-ld=bfd \
+              -Xlinker --add-needed -Xlinker --no-as-needed
diff -Nau wmake/rules/linux64Cray/c++Opt wmake/rules/linux64CrayToBeAdded/c++Opt
--- wmake/rules/linux64Cray/c++Opt	1970-01-01 02:00:00.000000000 +0200
+++ wmake/rules/linux64CrayToBeAdded/c++Opt	2025-02-25 13:28:52.000000000 +0200
@@ -0,0 +1,4 @@
+c++DBUG     =
+c++OPT      = -O3
+
+ROUNDING_MATH = -frounding-math
diff -Nau wmake/rules/linux64Cray/cOpt wmake/rules/linux64CrayToBeAdded/cOpt
--- wmake/rules/linux64Cray/cOpt	1970-01-01 02:00:00.000000000 +0200
+++ wmake/rules/linux64CrayToBeAdded/cOpt	2025-02-25 13:28:52.000000000 +0200
@@ -0,0 +1,2 @@
+cDBUG       =
+cOPT        = -O3
diff -Nau wmake/rules/linux64Cray/general wmake/rules/linux64CrayToBeAdded/general
--- wmake/rules/linux64Cray/general	1970-01-01 02:00:00.000000000 +0200
+++ wmake/rules/linux64CrayToBeAdded/general	2025-02-25 13:28:52.000000000 +0200
@@ -0,0 +1,6 @@
+PROJECT_LIBS = -l$(WM_PROJECT) -ldl
+
+include $(GENERAL_RULES)/standard
+
+include $(DEFAULT_RULES)/c
+include $(DEFAULT_RULES)/c++
