--- AdaptiveCpp/src/runtime/hip/hip_allocator_orig.cpp	2024-03-01 16:18:13.000000000 +0200
+++ AdaptiveCpp/src/runtime/hip/hip_allocator.cpp	2024-03-01 16:18:55.000000000 +0200
@@ -59,7 +59,7 @@
   void *ptr;
   hip_device_manager::get().activate_device(_dev);
 
-  hipError_t err = hipHostMalloc(&ptr, bytes, hipHostMallocDefault);
+  hipError_t err = hipHostMalloc(&ptr, bytes, hipHostMallocDefault | hipHostMallocNumaUser );
 
   if (err != hipSuccess) {
     register_error(__hipsycl_here(),
