diff --git a/setup.py b/setup.py
index 839a53a025..2f58eb2119 100644
--- a/setup.py
+++ b/setup.py
@@ -558,9 +558,9 @@ setup(name='gpaw',
           'console_scripts': ['gpaw = gpaw.cli.main:main']},
       setup_requires=['numpy'],
       python_requires=python_requires,
-      install_requires=[f'ase>={ase_version_required}',
-                        'numpy',
-                        'scipy>=1.6.0',
+      install_requires=['ase==3.25.0',
+                        'numpy==1.24.4',
+                        'scipy==1.10.1',
                         'gpaw-data'],
       extras_require={
           'docs': ['sphinx-rtd-theme',
