--- a/setup.py
+++ b/setup.py
@@ -395,7 +395,7 @@ setup(name='gpaw',
           'console_scripts': ['gpaw = gpaw.cli.main:main']},
       setup_requires=['numpy'],
       python_requires=python_requires,
-      install_requires=[f'ase>={ase_version_required}',
+      install_requires=[f'ase=={ase_version_required}',
                         'scipy>=1.6.0',
                         'pyyaml'],
       extras_require={'docs': ['sphinx-rtd-theme',
