diff -ruN spla-1.5.5.orig/src/gpu_util/gpu_pointer_translation.hpp spla-1.5.5/src/gpu_util/gpu_pointer_translation.hpp
--- spla-1.5.5.orig/src/gpu_util/gpu_pointer_translation.hpp	2023-04-25 09:55:56.000000000 +0300
+++ spla-1.5.5/src/gpu_util/gpu_pointer_translation.hpp	2024-08-22 16:57:26.000000000 +0300
@@ -54,7 +54,7 @@
   std::pair<const T*, const T*> ptrPair{nullptr, nullptr};
 
   // get memory type - cuda 10 changed attribute name
-#if defined(SPLA_CUDA) && (CUDART_VERSION >= 10000)
+#if (defined(SPLA_CUDA) && (CUDART_VERSION >= 10000)) || (defined(SPLA_ROCM) && HIP_VERSION_MAJOR > 5)
   auto memoryType = attr.type;
 #else
   auto memoryType = attr.memoryType;
