diff --git a/NAMD_3.0_Source/arch/Linux-x86_64.hip b/NAMD_3.0_Source/arch/Linux-x86_64.hip
index 98a2090..c177886 100644
--- a/NAMD_3.0.2_Source/arch/Linux-x86_64.hip
+++ b/NAMD_3.0.2_Source/arch/Linux-x86_64.hip
@@ -8,11 +8,17 @@ HIPDIR  = $(ROCMDIR)
 #endif
 
 ifeq ($(PLATFORM),)
-    PLATFORM = $(shell $(HIPDIR)/bin/hipconfig -P)
+    PLATFORM =amd#$(shell hipconfig -P)
 endif
 CUDADIR = $(HIPDIR)
 export HIP_PLATFORM=$(PLATFORM)
-HIPCCOPTS = -std=c++17 -O3 -fPIE -fexpensive-optimizations -ffast-math $(shell export HIP_PLATFORM=$(PLATFORM); $(HIPDIR)/bin/hipconfig --cpp_config) 
+HIPCCOPTS = -std=c++17 -O3 -fPIE -fexpensive-optimizations -ffast-math -D__HIP_PLATFORM_HCC__=/opt/rocm-6.3.4clang -D__HIP_PLATFORM_AMD__=amd
+
+#$(shell export HIP_PLATFORM=$(PLATFORM); $(HIPDIR)/bin/hipconfig --cpp_config) 
+
+
+
+
 ifneq ($(HIPCUB_DIR),)
     HIPCCOPTS += -I$(HIPCUB_DIR)
 endif
@@ -54,4 +60,4 @@ endif
 CUDAOBJS = $(CUDAOBJSRAW)
 CUDA = $(CUDAFLAGS) -I. $(shell export HIP_PLATFORM=$(PLATFORM); $(HIPDIR)/bin/hipconfig --cpp_config) 
 CUDACC = $(HIPCC)
-CUDACCOPTS = $(HIPCCOPTS) $(CUDAFLAGS)
+CUDACCOPTS = $(HIPCCOPTS) $(CUDAFLAGS) 
