diff -ruN magma-2.7.2.orig/include/magma_mangling_cmake.h magma-2.7.2/include/magma_mangling_cmake.h
--- magma-2.7.2.orig/include/magma_mangling_cmake.h	2023-08-25 16:15:30.000000000 +0300
+++ magma-2.7.2/include/magma_mangling_cmake.h	2024-08-23 08:03:31.000000000 +0300
@@ -1,7 +1,16 @@
-#ifndef MAGMA_MANGLING_CMAKE_H
-#define MAGMA_MANGLING_CMAKE_H
+#ifndef MAGMA_HEADER_INCLUDED
+#define MAGMA_HEADER_INCLUDED
 
-/* Empty file. See magma_mangling.h
- * When using CMake, this gets replaced by Fortran name mangling. */
+/* Mangling for Fortran global symbols without underscores. */
+#define MAGMA_GLOBAL(name,NAME) name##_
 
-#endif  // MAGMA_MANGLING_CMAKE_H
+/* Mangling for Fortran global symbols with underscores. */
+#define MAGMA_GLOBAL_(name,NAME) name##_
+
+/* Mangling for Fortran module symbols without underscores. */
+#define MAGMA_MODULE(mod_name,name, mod_NAME,NAME) __##mod_name##_MOD_##name
+
+/* Mangling for Fortran module symbols with underscores. */
+#define MAGMA_MODULE_(mod_name,name, mod_NAME,NAME) __##mod_name##_MOD_##name
+
+#endif
diff -ruN magma-2.7.2.orig/interface_hip/blas_c_v2.cpp magma-2.7.2/interface_hip/blas_c_v2.cpp
--- magma-2.7.2.orig/interface_hip/blas_c_v2.cpp	2023-08-25 16:16:50.000000000 +0300
+++ magma-2.7.2/interface_hip/blas_c_v2.cpp	2024-08-23 08:59:23.000000000 +0300
@@ -1851,14 +1851,18 @@
     #ifdef MAGMA_HAVE_HIP
         // TODO: remove fallback when hipblas provides this routine
         hipblasCtrmm(
-		    queue->hipblas_handle(),
-		    hipblas_side_const( side ),
+                    queue->hipblas_handle(),
+                    hipblas_side_const( side ),
                     hipblas_uplo_const( uplo ),
                     hipblas_trans_const( trans ),
                     hipblas_diag_const( diag ),
-		    int(m), int(n),
-		    (hipblasComplex*)&alpha, (const hipblasComplex*)dA, int(ldda),
-		    (hipblasComplex*)dB, int(lddb) );
+                    int(m), int(n),
+                    (hipblasComplex*)&alpha, (const hipblasComplex*)dA, int(ldda),
+                    (hipblasComplex*)dB, int(lddb)
+    #if (HIP_VERSION_MAJOR >= 6)
+                    , (hipblasComplex*)dB, int(lddb)
+    #endif
+                    );
     #else
         hipblasCtrmm(
                     queue->hipblas_handle(),
diff -ruN magma-2.7.2.orig/interface_hip/blas_d_v2.cpp magma-2.7.2/interface_hip/blas_d_v2.cpp
--- magma-2.7.2.orig/interface_hip/blas_d_v2.cpp	2023-08-25 16:16:50.000000000 +0300
+++ magma-2.7.2/interface_hip/blas_d_v2.cpp	2024-08-23 08:58:18.000000000 +0300
@@ -1851,14 +1851,18 @@
     #ifdef MAGMA_HAVE_HIP
         // TODO: remove fallback when hipblas provides this routine
         hipblasDtrmm(
-		    queue->hipblas_handle(),
-		    hipblas_side_const( side ),
+                    queue->hipblas_handle(),
+                    hipblas_side_const( side ),
                     hipblas_uplo_const( uplo ),
                     hipblas_trans_const( trans ),
                     hipblas_diag_const( diag ),
-		    int(m), int(n),
-		    (double*)&alpha, (const double*)dA, int(ldda),
-		    (double*)dB, int(lddb) );
+                    int(m), int(n),
+                    (double*)&alpha, (const double*)dA, int(ldda),
+                    (double*)dB, int(lddb)
+    #if (HIP_VERSION_MAJOR >= 6)
+                    , (double*)dB, int(lddb)
+    #endif
+                    );
     #else
         hipblasDtrmm(
                     queue->hipblas_handle(),
diff -ruN magma-2.7.2.orig/interface_hip/blas_s_v2.cpp magma-2.7.2/interface_hip/blas_s_v2.cpp
--- magma-2.7.2.orig/interface_hip/blas_s_v2.cpp	2023-08-25 16:16:50.000000000 +0300
+++ magma-2.7.2/interface_hip/blas_s_v2.cpp	2024-08-23 08:56:51.000000000 +0300
@@ -1851,14 +1851,18 @@
     #ifdef MAGMA_HAVE_HIP
         // TODO: remove fallback when hipblas provides this routine
         hipblasStrmm(
-		    queue->hipblas_handle(),
-		    hipblas_side_const( side ),
+                    queue->hipblas_handle(),
+                    hipblas_side_const( side ),
                     hipblas_uplo_const( uplo ),
                     hipblas_trans_const( trans ),
                     hipblas_diag_const( diag ),
-		    int(m), int(n),
-		    (float*)&alpha, (const float*)dA, int(ldda),
-		    (float*)dB, int(lddb) );
+                    int(m), int(n),
+                    (float*)&alpha, (const float*)dA, int(ldda),
+                    (float*)dB, int(lddb)
+    #if (HIP_VERSION_MAJOR >= 6)
+                    , (float*)dB, int(lddb)
+    #endif
+                    );
     #else
         hipblasStrmm(
                     queue->hipblas_handle(),
diff -ruN magma-2.7.2.orig/interface_hip/blas_z_v2.cpp magma-2.7.2/interface_hip/blas_z_v2.cpp
--- magma-2.7.2.orig/interface_hip/blas_z_v2.cpp	2023-08-25 16:15:56.000000000 +0300
+++ magma-2.7.2/interface_hip/blas_z_v2.cpp	2024-08-23 08:24:23.000000000 +0300
@@ -1851,14 +1851,18 @@
     #ifdef MAGMA_HAVE_HIP
         // TODO: remove fallback when hipblas provides this routine
         hipblasZtrmm(
-		    queue->hipblas_handle(),
-		    hipblas_side_const( side ),
+                    queue->hipblas_handle(),
+                    hipblas_side_const( side ),
                     hipblas_uplo_const( uplo ),
                     hipblas_trans_const( trans ),
                     hipblas_diag_const( diag ),
-		    int(m), int(n),
-		    (hipblasDoubleComplex*)&alpha, (const hipblasDoubleComplex*)dA, int(ldda),
-		    (hipblasDoubleComplex*)dB, int(lddb) );
+                    int(m), int(n),
+                    (hipblasDoubleComplex*)&alpha, (const hipblasDoubleComplex*)dA, int(ldda),
+                    (hipblasDoubleComplex*)dB, int(lddb)
+    #if (HIP_VERSION_MAJOR >= 6)
+                    , (hipblasDoubleComplex*)dB, int(lddb)
+    #endif
+                    );
     #else
         hipblasZtrmm(
                     queue->hipblas_handle(),
diff -ruN magma-2.7.2.orig/interface_hip/interface.cpp magma-2.7.2/interface_hip/interface.cpp
--- magma-2.7.2.orig/interface_hip/interface.cpp	2023-08-25 16:15:53.000000000 +0300
+++ magma-2.7.2/interface_hip/interface.cpp	2024-08-23 08:47:02.000000000 +0300
@@ -213,7 +213,7 @@
                     g_magma_devices[dev].cuda_arch       = prop.major*100 + prop.minor*10;
                     g_magma_devices[dev].shmem_multiproc = prop.sharedMemPerMultiprocessor;
                     #elif defined(MAGMA_HAVE_HIP)
-                    g_magma_devices[dev].cuda_arch       = prop.gcnArch;
+                    g_magma_devices[dev].cuda_arch       = prop.major*100 + prop.minor*10;
                     g_magma_devices[dev].shmem_multiproc = prop.maxSharedMemoryPerMultiProcessor;
                     #endif
 
@@ -459,12 +459,12 @@
         #endif
 
         #ifdef MAGMA_HAVE_HIP
-        printf( "%% device %d: %s, %.1f MHz clock, %.1f MiB memory, gcn arch %d\n",
+        printf( "%% device %d: %s, %.1f MHz clock, %.1f MiB memory, gcn arch %s\n",
                 dev,
                 prop.name,
                 prop.clockRate / 1000.,
                 prop.totalGlobalMem / (1024.*1024.),
-                prop.gcnArch );
+                prop.gcnArchName );
         #endif
     }
 
@@ -473,7 +473,7 @@
     printf( "%% %s", ctime( &t ));
 }
 
-#if CUDA_VERSION >= 11000
+#if CUDA_VERSION >= 11000 || HIP_VERSION_MAJOR <= 6
 #define magma_memoryType() type
 #else
 #define magma_memoryType() memoryType
@@ -528,7 +528,11 @@
                   #endif
 
                 #elif defined(MAGMA_HAVE_HIP)
-                return (attr.memoryType == hipMemoryTypeDevice);
+                  #if HIP_VERSION_MAJOR >= 6
+                    return (attr.type == hipMemoryTypeDevice);
+                  #else
+                    return (attr.memoryType == hipMemoryTypeDevice);
+                  #endif
                 #endif
             }
             else if ( err == hipErrorInvalidValue ) {
diff -ruN magma-2.7.2.orig/sparse_hip/include/magmasparse_types.h magma-2.7.2/sparse_hip/include/magmasparse_types.h
--- magma-2.7.2.orig/sparse_hip/include/magmasparse_types.h	2023-08-25 16:15:54.000000000 +0300
+++ magma-2.7.2/sparse_hip/include/magmasparse_types.h	2024-08-23 09:04:01.000000000 +0300
@@ -21,7 +21,7 @@
 #endif
 
 // includes CUDA
-#include <hipsparse.h>
+#include <hipsparse/hipsparse.h>
 
 /* (author: Cade Brown <cbrow216@vols.utk.edu>
  *
